src/app/shared/services/task/task.service.ts

```diff
@@ -200,6 +221,126 @@ export class TaskService extends AsyncServiceBase {
     });
   }
 
+  private registerTemplateActionHandlers() {
+    this.templateActionRegistry.register({
+      task: async ({ args, params }) => {
+        const [actionId] = args;
+        const childActions = {
+          /**
+           * HACK: evaluate completion status of task groups. Pending an overhaul of the task system,
+           * this action offers a way to evaluate task group completion status over multiple nested task groups.
+           *
+           * @param {string} task_groups A space-separated list of task group references in the format "parentDataListName.rowId",
+           * starting with the highest level task group and working down to the lowest level.
+           * The final entry is just a data list name corresponding the the list of subtasks of the last child task group
+           */
+          evaluate_group_completion: async () => {
+            if (!params.task_groups)
+              return console.warn(
+                "[TASK] evaluate_group_completion action - To evaluate task group completion, a list of task group references must be provided via the task_groups param"
+              );
+            const taskGroupRefs = params.task_groups.split(" ");
+            const nestedTaskGroups = this.parseTaskGroupRefsToTaskGroups(taskGroupRefs);
+            console.log(
+              "[TASK] Debug - evaluating task group completion status. Nested task groups object:",
+              nestedTaskGroups
```

Comment:
I've left this log in for now as I think it could be useful for debugging when authoring the action (especially with task group references being read from complex local variables)